tests/lib/ocpp/v201/test_smart_charging_handler.cpp

```diff
@@ -1168,4 +1208,233 @@ TEST_F(ChargepointTestFixtureV201, K01_ValidateAndAdd_AddsValidProfiles) {
     EXPECT_THAT(profiles, testing::Contains(profile));
 }
 
+TEST_F(ChargepointTestFixtureV201, K09_GetChargingProfiles_EvseId) {
+    auto periods = create_charging_schedule_periods({0, 1, 2});
+
+    auto profile1 = create_charging_profile(
+        1, ChargingProfilePurposeEnum::ChargingStationMaxProfile,
+        create_charge_schedule(ChargingRateUnitEnum::A, periods, ocpp::DateTime("2024-01-17T17:00:00")));
+    auto profile2 = create_charging_profile(
+        2, ChargingProfilePurposeEnum::TxDefaultProfile,
+        create_charge_schedule(ChargingRateUnitEnum::A, periods, ocpp::DateTime("2024-01-17T17:00:00")));
+
+    auto sut1 = handler.validate_and_add_profile(profile1, 0);
+    auto sut2 = handler.validate_and_add_profile(profile2, DEFAULT_EVSE_ID);
+
+    auto profiles = handler.get_profiles();
+    EXPECT_THAT(profiles, testing::SizeIs(2));
+
+    auto reported_profiles = handler.get_profiles(
+        create_get_charging_profile_request(DEFAULT_REQUEST_ID, create_charging_profile_criteria(), 0));
```

Comment:
You can use the constants for the EVSE ID when calling create_get_charging_profile_request() in the tests to make it clear how this line ties back to the setup of the tests.